nibabel/loadsave.py

```diff
@@ -20,6 +20,54 @@
 from .deprecated import deprecate_with_version
 
 
+def _signature_matches_extension(filename, sniff):
+    """Check if signature aka magic number matches filename extension.
+
+    Parameters
+    ----------
+    filename : str or os.PathLike
+        Path to the file to check
+
+    sniff : bytes or None
+        First bytes of the file. If not `None` and long enough to contain the
+        signature, avoids having to read the start of the file.
+
+    Returns
+    -------
+    matches : bool
+       - `True` if the filename extension is not recognized (not .gz nor .bz2)
+       - `True` if the magic number was successfully read and corresponds to
+         the format indicated by the extension.
+       - `False` otherwise.
+    error_message : str
+       An error message if opening the file failed or a mismatch is detected;
+       the empty string otherwise.
+
+    """
+    signatures = {
+        ".gz": {"signature": b"\x1f\x8b", "format_name": "gzip"},
+        ".bz2": {"signature": b"\x42\x5a\x68", "format_name": "bzip2"}
```

Comment:
Any reason to use hex codes instead of ASCII values here?

```suggestion
        ".bz2": {"signature": b"BZh", "format_name": "bzip2"}
```

Reply:
no reason!